### PR TITLE
Add workflow to submit GH dependency graph

### PR DESCRIPTION
**Pull request by bigdaz**
This workflow will calculate and submit a GitHub Dependency Graph for any change to the 'main' branch. This allows GitHub to display the dependencies in the "Insights" tab, and to inspect all dependencies for security vulnerabilities.

### CODE REVIEW

File: .github/workflows/submit-github-dependency-graph.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      with:
+        distribution: temurin
+        java-version: 8
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2    
+    - name: Generate dependency graph
+      uses: gradle/gradle-build-action/actions/dependency-graph-generate@v2
+      env:
+        GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
+    - name: Submit dependency graph
```
Comment (jprinet) [inline .github/workflows/submit-github-dependency-graph.yml:25]: Just curious if you have considered merging the generate and submit steps?
is this for troubleshooting reasons or that other sinks could be used?

**Review by jprinet — APPROVED** [review]
LGTM

**Comment by StefMa** [comment]
In case you would use Gradle version catalog this would happen
automatically, right? 🤔

File: .github/workflows/submit-github-dependency-graph.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    - uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 8
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2    
+    - name: Generate dependency graph
+      uses: gradle/gradle-build-action/actions/dependency-graph-generate@v2
+      env:
+        GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
```
Comment (runningcode) [inline .github/workflows/submit-github-dependency-graph.yml:24]: is this needed to publish a build scan?

**Review by runningcode — APPROVED** [review]

**Comment by runningcode** [comment]
@StefMa No, this is not possible there is not enough information in the version catalog to be able to do this. I suspect you might have a different understanding of the goals of this PR though. Please ask further in slack!

File: .github/workflows/submit-github-dependency-graph.yml (hunk 2 above)
Comment (bigdaz) [reply .github/workflows/submit-github-dependency-graph.yml:24]: That's my assumption, based on experience with CCUD and cache-fix plugin projects.

File: .github/workflows/submit-github-dependency-graph.yml (hunk 1 above)
Comment (bigdaz) [reply .github/workflows/submit-github-dependency-graph.yml:25]: The main reason for the separation is that it's necessary for PR builds, where the "submit" step has to be done in a separate workflow with sufficient permissions.

**Pull request merged**